observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemRegistrar.java

```diff
@@ -0,0 +1,213 @@
+                .addAttributes(List.of(ENDPOINT, STEP, EXPOSED_SUBSYSTEMS))
+                .withDeploymentChainContributor(target -> {
+                    target.addDeploymentProcessor(MicrometerConfigurationConstants.NAME, DEPENDENCIES, DEPENDENCIES_MICROMETER,
+                            new MicrometerDependencyProcessor());
+                    target.addDeploymentProcessor(MicrometerConfigurationConstants.NAME, POST_MODULE, POST_MODULE_MICROMETER,
+                            new MicrometerDeploymentProcessor(deploymentConfig.get()));
+                })
+                .withAddOperationRestartFlag(OperationEntry.Flag.RESTART_ALL_SERVICES)
+                .withRemoveOperationRestartFlag(OperationEntry.Flag.RESTART_ALL_SERVICES)
+                .build();
+
+        ManagementResourceRegistrar.of(descriptor).register(registration);
+
+        return registration;
+    }
+
+    @Override
+    public ResourceServiceInstaller configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        List<String> exposedSubsystems = MicrometerSubsystemRegistrar.EXPOSED_SUBSYSTEMS.unwrap(context, model);
+        boolean exposeAnySubsystem = exposedSubsystems.remove("*");
+        String endpoint = MicrometerSubsystemRegistrar.ENDPOINT.resolveModelAttribute(context, model).asStringOrNull();
+        Long step = MicrometerSubsystemRegistrar.STEP.resolveModelAttribute(context, model).asLong();
+
+        WildFlyRegistry wildFlyRegistry = endpoint != null ?
+                new WildFlyOtlpRegistry(new WildFlyMicrometerConfig(endpoint, step)) :
+                new NoOpRegistry();
+
+        try {
+            new JmxMicrometerCollector(wildFlyRegistry).init();
+        } catch (IOException e) {
+            throw MICROMETER_LOGGER.failedInitializeJMXRegistrar(e);
+        }
+
+        ServiceDependency<ModelControllerClientFactory> mccf = ServiceDependency.on(CLIENT_FACTORY_CAPABILITY, ModelControllerClientFactory.class);
+        ServiceDependency<Executor> executor = ServiceDependency.on(MANAGEMENT_EXECUTOR, Executor.class);
+        ServiceDependency<ProcessStateNotifier> processStateNotifier = ServiceDependency.on(PROCESS_STATE_NOTIFIER, ProcessStateNotifier.class);
+
+        Supplier<MicrometerCollector> collectorSupplier = () ->
+                new MicrometerCollector(mccf.get().createClient(executor.get()), processStateNotifier.get(), wildFlyRegistry);
+
+        deploymentConfig.set(new MicrometerDeploymentConfiguration() {
+            @Override
+            public WildFlyRegistry getRegistry() {
+                return wildFlyRegistry;
+            }
+
+            @Override
+            public Predicate<String> getSubsystemFilter() {
+                return subsystem -> exposeAnySubsystem || exposedSubsystems.contains(subsystem);
+            }
+
+            @Override
+            public Supplier<MicrometerCollector> getCollectorSupplier() {
+                return collectorSupplier;
+            }
```

Comment:
This will result in a 2nd Micrometer collector instance being created by your DUP.  I doubt that was your intention.
Either change this method to return `captor::get`; or, better yet, change this method signature to `MicrometerCollector getCollector()` and have this method return `captor.get()`

Reply:
On 2nd read, it seems to me you can remove this method all together.